autoscheduler/frontend/src/components/SchedulingPage/CourseSelectColumn/CourseSelectCard/ExpandedCourseCard/SectionSelect/SectionSelect.tsx

```diff
@@ -33,174 +24,24 @@ const SectionSelect: React.FC<SectionSelectProps> = ({ id }): JSX.Element => {
-  };
-
-  /**
-   * Accepts an array of meetings and returns a filtered array without duplicate meetings.
-   * Meetings are considered to be duplicates if they are of the same type, meet on the same days,
-   * and start at the same time. Meetings that are the same by all of these criteria but
-   * differ only in the end times will still be considered duplicates
-   * @param arr
-   */
-  const filterDuplicateMeetings = (arr: Meeting[]): Meeting[] => {
-    // helper function to merge two meetings
-    const mergeMeetings = (mtg1: Meeting, mtg2: Meeting): Meeting => {
-      if (!mtg2) return mtg1;
-
-      // choose the later end time
-      const [laterEndHours, laterEndMinutes] = mtg2.endTimeHours > mtg1.endTimeHours
-        ? [mtg2.endTimeHours, mtg2.endTimeMinutes]
-        : [mtg1.endTimeHours, mtg1.endTimeMinutes];
-      // merge the days array by logical OR of each element
-      const days = mtg1.meetingDays.map((hasMeeting, idx) => hasMeeting || mtg2.meetingDays[idx]);
-      return {
-        ...mtg1,
-        endTimeHours: laterEndHours,
-        endTimeMinutes: laterEndMinutes,
-        meetingDays: days,
-      };
-    };
-
-    // add all meetings to a map, then get the values of the map
-    const uniqueMeetings = new Map<string, Meeting>();
-    arr.forEach((mtg) => {
-      const key = `${mtg.meetingType}${mtg.startTimeHours}${mtg.startTimeMinutes}`;
-      uniqueMeetings.set(key, mergeMeetings(mtg, uniqueMeetings.get(key)));
-    });
-    return [...uniqueMeetings.values()];
-  };
-
-  // returns a div containing the section's number and available/max enrollment
-  const createSectionHeader = (section: Section): JSX.Element => {
-    const remainingSeats = section.maxEnrollment - section.currentEnrollment;
-    const remainingSeatsColor = remainingSeats > 0 ? 'black' : 'red';
-    // show section number and remaining seats if this is the first meeting for a section
-    return (
-      <Typography className={styles.denseListItem} component="tr">
-        <td>
-          {section.sectionNum}
-        </td>
-        <td
-          style={{ color: remainingSeatsColor, textAlign: 'right' }}
-          colSpan={3}
-        >
-          {`${remainingSeats}/${section.maxEnrollment} seats left`}
-        </td>
-      </Typography>
-    );
-  };
-
-  const renderMeeting = (mtg: Meeting, showSectionNum: boolean): JSX.Element => (
-    <React.Fragment key={mtg.id}>
-      {showSectionNum ? createSectionHeader(mtg.section) : null }
-      <Typography className={styles.denseListItem} color="textSecondary" component="tr">
-        <td>{MeetingType[mtg.meetingType]}</td>
-        <td>{mtg.building || 'ONLINE'}</td>
-        <td>{formatMeetingDays(mtg)}</td>
-        <td>{getMeetingTimeText(mtg)}</td>
-      </Typography>
-    </React.Fragment>
-  );
-
   const makeList = (): JSX.Element[] => {
     let lastProf: string = null;
     let lastHonors = false;
-    return sections.map(({ section, selected, meetings }, secIdx) => {
```

Comment:
I'm so glad this is a separate component now, no offense but this was an absolute mess before lol